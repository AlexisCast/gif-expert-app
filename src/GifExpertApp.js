import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ({defaultCategories=[]}) => {
	// const [categories, setCategories] = useState(["Dragon Ball"]);
	const [categories, setCategories] = useState(defaultCategories);

	return (
		<>
			<h2 className="title animate__animated animate__fadeIn">GifExpertApp</h2>

			<AddCategory setCategories={setCategories} />

			<hr />

			<ol>
				{categories.map((category) => (
					<GifGrid key={category} category={category} />
				))}
			</ol>
			<footer>
				<p className="title">
					Alexis Castellanos
				</p>
			</footer>
		</>
	);
};

export default GifExpertApp;
