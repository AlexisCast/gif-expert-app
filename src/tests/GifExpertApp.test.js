import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";


describe('Pruebas em <GifExpertApp',()=>{
   test('debe de mostrar correctamente',()=>{
      const wrapper=shallow(<GifExpertApp/>);
      expect(wrapper).toMatchSnapshot();
   });
   test('debe de mostrar una vista de categoris',()=>{
      const categories=['One Punch','Dragon ball'];
      const wrapper=shallow(<GifExpertApp defaultCategories={categories}/>);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('GifGrid').length).toBe(categories.length);
   })
})