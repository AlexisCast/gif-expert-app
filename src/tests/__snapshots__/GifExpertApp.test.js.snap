// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas em <GifExpertApp debe de mostrar correctamente 1`] = `
<Fragment>
  <h2
    className="title animate__animated animate__fadeIn"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
  <footer>
    <p
      className="title"
    >
      Alexis Castellanos
    </p>
  </footer>
</Fragment>
`;

exports[`Pruebas em <GifExpertApp debe de mostrar una vista de categoris 1`] = `
<Fragment>
  <h2
    className="title animate__animated animate__fadeIn"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
  <footer>
    <p
      className="title"
    >
      Alexis Castellanos
    </p>
  </footer>
</Fragment>
`;
